new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CleanupUtil.java

```diff
@@ -0,0 +1,524 @@
+
+      for (var namespace : namespaces) {
+        logger.info("Check for artifacts in namespace {0}", namespace);
+        withStandardRetryPolicy
+            .conditionEvaluationListener(
+                condition -> logger.info("Waiting for artifacts to be deleted in namespace {0}, "
+                    + "(elapsed time {1} , remaining time {2}",
+                    namespace,
+                    condition.getElapsedTimeInMS(),
+                    condition.getRemainingTimeInMS()))
+            .until(artifactsDoesntExist(namespace));
+      }
+    } catch (Exception ex) {
+      logger.warning(ex.getMessage());
+      logger.warning("Cleanup failed");
+    }
+  }
+
+  /**
+   * Delete domains in the given namespace, if exists.
+   *
+   * @param namespace name
+   */
+  private static void deleteDomains(String namespace) {
+    try {
+      for (var item : Kubernetes.listDomains(namespace).getItems()) {
+        String domainUid = item.getMetadata().getName();
+        Kubernetes.deleteDomainCustomResource(domainUid, namespace);
+      }
+    } catch (Exception ex) {
+      logger.severe(ex.getMessage());
+      logger.severe("Failed to delete domain or the {0} is not a domain namespace", namespace);
+    }
+  }
+
+  /**
+   * Uninstalls the operator.
+   *
+   * @param namespace name
+   */
+  private static void uninstallOperator(String namespace) {
+    HelmParams opHelmParams = new HelmParams()
+        .releaseName(TestConstants.OPERATOR_RELEASE_NAME)
+        .namespace(namespace);
+    TestActions.uninstallOperator(opHelmParams);
+  }
+
+  /**
+   * Returns true if the namespace is operator namespace, otherwise false.
+   *
+   * @param namespace name
+   * @return true if the namespace is operator namespace, otherwise false
+   * @throws ApiException when Kubernetes cluster query fails
+   */
+  public static boolean isOperatorNamespace(String namespace) throws ApiException {
+    return !Kubernetes.listPods(namespace, "weblogic.operatorName").getItems().isEmpty();
+  }
+
+  /**
+   * Returns true if artifacts doesn't exists in the Kubernetes cluster.
+   *
+   * @param namespace name
+   * @return true if no artifacts exists otherwise false
+   */
+  public static Callable<Boolean> artifactsDoesntExist(String namespace) {
+    return () -> {
+      boolean doesnotExist = true;
+      logger.info("Checking for artifacts in namespace {0}\n", namespace);
+
+      // Check if domain exists
+      try {
+        if (!Kubernetes.listDomains(namespace).getItems().isEmpty()) {
+          logger.info("Domain still exists !!!");
+          List<Domain> items = Kubernetes.listDomains(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list domains");
+      }
+
+      // Check if the replica sets exist
```

Comment:
check if any replica sets exist (and so on through the rest of this method)

Reply:
fixed